docs/DeveloperGuide.md

```diff
@@ -1,34 +1,851 @@
+This section describes some noteworthy details on how certain features are being implemented.
+
+### 4.1 Store Data <a name="4.1"></a>
+The storage feature saves the data of the user so that 
+the ModTracker application continues from where the user left off the previous time. 
+
+#### Current Implementation 
+The `Storage` class facilitates the data saving and loading mechanism. 
+The constructor `Storage(filePath)` accepts a String which is the file path to an external file.
+This external file stores the user's data locally.
+
+Inside the external file, it stores the name of the user 
+followed by any valid inputs which modifies the data.
+The format of the data inside the external file is as follows:
+
+````
+username
+valid user input 1
+valid user input 2
+valid user input 3
+...
+````
+
+When the user runs the program again, 
+the program will first load data from the external file.
+The `Parser` class then parses these data 
+before the program prompts the user for input.
+
+Given below is an example usage scenario, showing the 2 different times 
+when a user starts the application.
+
+##### First Use
+The user launches the application for the first time.
+
+1. `ModTracker` creates a new `Storage` object 
+with file path at `data/modtracker.txt`. 
+
+1. The newly created `Storage` object checks that there is no file 
+at the specified file path, and creates a new file there.
+
+1. The program prompts the user to input his/ her name.
+
+1. The `Storage` object writes this username into the external file.
+
+1. The program prompts the user for further inputs.
+
+1. The user enters `help`, and the program displays the help message. 
+Since this command does not modify the data, 
+the external file remains unchanged.
+
+1. The user enters `addmod CS1010 10`, 
+and the program adds the new module correspondingly. 
+Since this is a valid command which modifies the data,
+ the `Storage` object appends `addmod CS1010 10` to the external file.
+
+1. The user enters `exit`, and the program terminates.
+
+##### Second Use
+The same user starts the application again (on the same device as the first use).
+
+1. `ModTracker` creates a new `Storage` object 
+with the same specified file path at `data/modtracker.txt`.
+   
+1. The `Storage` object checks that the file at `data/modtracker.txt` exists,
+and reads the first line of the file to obtain the user's name.
+
+1. The `Storage` object passes the username to `ModTracker`, 
+which calls `Ui` to greet the user.
+
+1. `ModTracker` then obtains the remaining lines in the file via `Storage`,
+and passes it to `Parser` to load the data into `ModuleList` and `TaskList`.
+
+1. The program prompts the user for further inputs,
+and it continues as per normal, with the data loaded.
+
+The following sequence diagram shows how the storage feature works,
+with a focus on how it obtains the username:
+
+![Storage Sequence Diagram](diagrams/Storage%20Sequence%20Diagram.svg)
```

Comment:
Is there supposed to be an overlap here? Could you lower the constructor bar to avoid this overlap?
![review1](https://user-images.githubusercontent.com/57140494/97385758-6124c100-190d-11eb-80aa-3a89b1e0e52b.png)